internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -758,39 +759,93 @@ func (md *machineDeployment) updateUsingRollingStrategy(parentCtx context.Contex
 
 	for group, entries := range entriesByGroup {
 		entries := entries
-		startIdx += len(entries)
+
+		warmMachines := lo.Filter(entries, func(e *machineUpdateEntry, i int) bool {
+			return e.leasableMachine.Machine().State == "started"
+		})
+		coldMachines := lo.Filter(entries, func(e *machineUpdateEntry, i int) bool {
+			return e.leasableMachine.Machine().State != "started"
+		})
+
 		groupsPool.Go(func(ctx context.Context) error {
-			return md.updateEntriesGroup(ctx, group, entries, sl, startIdx-len(entries))
+			errChan := make(chan error, 2)
+			wg := sync.WaitGroup{}
+
+			wg.Add(1)
+			go func() {
```

Comment:
Wouldn't errgroup instead of simple go routine+ wait group be useful here in terms of conciseness since we'll already using it anyway? I'm also wondering if there are not side effects of running go routines inside go routines.

Reply:
> I'm also wondering if there are not side effects of running go routines inside go routines.
func main() is a goroutine.